src/api/firebase/auth.ts

```diff
@@ -0,0 +1,32 @@
+import { createUserWithEmailAndPassword, signInWithEmailAndPassword } from 'firebase/auth';
+import { auth } from './firebase';
+import { CreateUserData, LoginUserData } from '../../utils/interfaces';
+
+// Create a new user and automatically login the new registered user
+export const createUser = (
+  email: string,
+  password: string,
+  formData: CreateUserData,
+  setUserCredentials: CallableFunction
+) => {
+  createUserWithEmailAndPassword(auth, email, password)
+    .then((userCredentials) => {
+      setUserCredentials(formData);
+      console.log(userCredentials.user);
+      console.log(formData);
+    })
+    .catch((error) => {
+      console.log(error.message);
+    });
+};
+
+// Login an existing user
+export const logInUser = (formData: LoginUserData, setUserCredentials: CallableFunction) => {
```

Comment:
Ar trebui sa fie un SLICE, in store, cu REDUX THUNK pentru a salva automat datele sesiunii. Incearca sa afli exact ce returneaza loginu. Cookies, localStorage, sessionStorage.

Referinta, globalState slice.

Reply:
Am facut o incercare de thrunk dau commit imediat.

Reply:
Foloseste await/async

Slice-urile sunt in store/slices. uitate mai atent la ce am facut eu.